services/src/main/java/org/fao/geonet/api/es/EsHTTPProxy.java

```diff
@@ -646,15 +646,22 @@ private void processResponse(ServiceContext context, HttpSession httpSession,
                     addRelatedTypes(doc, relatedTypes, context);
                 }
 
-                if (doc.has("_source")) {
+                if (doc.has("_source") &&
+                    doc.get("_source").has(Geonet.IndexFieldNames.SCHEMA)) {
                     ObjectNode sourceNode = (ObjectNode) doc.get("_source");
 
-                    String metadataSchema = doc.get("_source").get("documentStandard").asText();
-                    MetadataSchema mds = schemaManager.getSchema(metadataSchema);
-
-                    // Apply metadata schema filters to remove non-allowed fields
-                    processMetadataSchemaFilters(context, mds, doc);
-
+                    String metadataSchema = doc.get("_source").get(Geonet.IndexFieldNames.SCHEMA).asText();
+                    try {
+                        MetadataSchema mds = schemaManager.getSchema(metadataSchema);
+
+                        // Apply metadata schema filters to remove non-allowed fields
+                        processMetadataSchemaFilters(context, mds, doc);
+                    } catch (IllegalArgumentException e) {
+                        LOGGER.error("Failed to load metadata schema for {}. Error is: {}",
+                            getSourceString(doc, Geonet.IndexFieldNames.UUID),
+                            e.getMessage()
+                        );
+                    }
                     // Remove fields with privileges info
```

Comment:
The code change affects also this code that doesn't depend on `Geonet.IndexFieldNames.SCHEMA`